components/resources/library/src/macosMain/kotlin/org/jetbrains/compose/resources/ResourceEnvironment.macos.kt

```diff
@@ -0,0 +1,32 @@
+package org.jetbrains.compose.resources
+
+import kotlinx.cinterop.*
+import platform.AppKit.NSScreen
+import platform.CoreGraphics.CGDisplayPixelsWide
+import platform.CoreGraphics.CGDisplayScreenSize
+import platform.Foundation.*
+
+internal actual fun getResourceEnvironment(): ResourceEnvironment {
+    val locale = NSLocale.currentLocale()
+    val isDarkTheme = NSUserDefaults.standardUserDefaults.stringForKey("AppleInterfaceStyle") == "Dark"
+
+    val dpi = NSScreen.mainScreen?.let { screen ->
+        val backingScaleFactor = screen.backingScaleFactor
+        val screenNumber = interpretObjCPointer<NSNumber>(
+            screen.deviceDescription["NSScreenNumber"].objcPtr()
+        ).unsignedIntValue
+
+        val displaySizePX = CGDisplayPixelsWide(screenNumber).toFloat() * backingScaleFactor
+        val displaySizeMM = CGDisplayScreenSize(screenNumber).useContents { width }
+
+        //1 inch = 25.4 mm
```

Comment:
The same. It's better to have named constant or explicit conversion function

Reply:
why?

Reply:
there is no a reason to have such small granularity

Reply:
Function should be easy readable. Granularity and cleanness are tools for achieve that, not otherwise